packages/gatsby/src/schema/__tests__/create-key.js

```diff
@@ -0,0 +1,26 @@
+const createKey = require(`../create-key`)
+
+describe(`createKey`, () => {
+  it(`leaves valid strings as is`, () => {
+    ;[`01234`, `validstring`, `_hello`, `_`].forEach(input => {
+      expect(createKey(input)).toBe(input)
+    })
+  })
+
+  it(`replaces invalid characters`, () => {
+    ;[
+      [`/hello`, `_hello`],
+      [`~/path/to/some/module`, `_-path-to-some-module`],
+      [`/*`, `_-`],
+      [`/*.js`, `_--js`],
+    ].forEach(([input, output]) => {
+      expect(createKey(input)).toBe(output)
+    })
+  })
+
+  it(`does not generate same key for different input`, () => {
+    ;[[`/*.js`, `*js`]].forEach(([one, two]) => {
```

Comment:
Could probably _not_ have written this as an array since I'm only testing one sample input. Oh well 🤷‍♂️